### PR TITLE
Add spider for Campos dos Goytacazes/RJ

### PR DESCRIPTION
**Pull request by chamun**
This PR continues the work started in #70 as part of #221.

It addresses the change requests from #70, merges main into the branch and fixes the code to download all the gazettes since 2015.

### CODE REVIEW

**Comment by chamun** [comment]
The failing CI error is unrelated to this PR. I have opened https://github.com/okfn-brasil/querido-diario/pull/236 to address this problem.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:20]: **suggestion**: If you are creating a `datetime.date` object, there is no need to use an intermediary like `dateparser`.

[Built-in date](https://docs.python.org/3/library/datetime.html#datetime.date) from Python is enough:

`LIMIT_DATE = datetime.date(2015, 1, 1)`

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:12]: **suggestion**: Considering that we only use this selector once, and we have many other selectors inside `parse` method, in my opinion having this here is not helpful. If we need to change it because the layout changed, it is very likely that we will have to update the entire spider anyway.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:31]: **suggestion** Scrapy suggest the use of `.get` instead of (deprecated) `extract_first`. 

**suggestion** `get` (and `extract_first`) accepts a `default` [argument](https://parsel.readthedocs.io/en/latest/parsel.html#parsel.selector.SelectorList.get), so you don't need to add this `or ""`

This is equivalent to:
`gazette_text = element.css("h4::text").get("")`

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:33]: **suggestion** If you are using `gazette_text` just to apply a regex over it, you can use `re` method:

This will return a tuple with 3 items containing the information you want:
`date = element.css("h4::text").re('r"((\d{2}) de (.*) de (\d{4})')`
It will return `None` if it didn't match, so you can check it as you did it next.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:38]: **praise** Good catch :-) This kind of error is difficult to find sometimes.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:41]: **question** Why you are limiting the date? If we have older gazettes, we should get it as well.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:45]: **suggestion** If the URL is relative, but the netloc is still the same, you can use `response.urljoin` to make the absolute URL:
`url = response.urljoin(element.css("a::attr(href)").get())`

**Review by rennerocha — CHANGES_REQUESTED** [review]

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (chamun) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:41]: Because [CONTRIBUTING.md](https://github.com/okfn-brasil/querido-diario/blob/main/CONTRIBUTING.md) states that the projec'ts goal is  to download gazettes since 2015:

> At this moment, the project aims for the 100 top Brazilian municipalities by population, collecting all the gazettes since 2015 

Campos Goytacazes City Hall's website provides gazettes up until 2010.

Should I remove this limit?

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (chamun) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:45]: Nice!

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (chamun) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:33]: `gazette_text` is also used to set the value of `is_extra_edition`.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (chamun) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:20]: I'll address this once we settle on https://github.com/okfn-brasil/querido-diario/pull/235#discussion_r491440601.

**Comment by chamun** [comment]
@rennerocha this PR is ready for another look.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (jvanz) [inline processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:8]:
```suggestion
from datetime import datetime
import re

import dateparser
from scrapy import Request
from scrapy.exceptions import CloseSpider

from gazette.items import Gazette
from gazette.spiders.base import BaseGazetteSpider
```

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (jvanz) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:19]: I think you forget to update this line. Because I cannot see the change in the highlighted comment. Am I wrong?

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (jvanz) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:40]: Yes. Please, get all the files available. I think what that line was trying to say is that the spiders in the time of that writing was crawling files since 2015.

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (jvanz) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:37]: :+1:

**Review by jvanz — CHANGES_REQUESTED** [review]

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (chamun) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:19]: I did not update this line because I was waiting on a reply on my question in https://github.com/okfn-brasil/querido-diario/pull/235#discussion_r491440601. I'll update it now!

File: processing/data_collection/gazette/spiders/rj_campos_goytacazes.py
Comment (chamun) [reply processing/data_collection/gazette/spiders/rj_campos_goytacazes.py:40]: @jvanz OK, I'll work on that!

**Comment by chamun** [comment]
@jvanz @rennerocha All the change requests have been addressed.

**Review by jvanz — APPROVED** [review]
LGTM. 

For the maintainer who merge the PR. Please, use "squash and merge option" in the web interface.

**Pull request merged**
